TombEngine/Game/control/trigger.cpp

```diff
@@ -175,6 +175,41 @@ bool SwitchTrigger(short itemNumber, short timer)
 	if (item.ObjectNumber >= ID_KEY_HOLE1 && item.ObjectNumber <= ID_KEY_HOLE16)
 		return false;
 
+	//Handle Pulley
+	if (item.ObjectNumber == ID_PULLEY &&
```

Comment:
This function becomes more and more hacky. Can we move pulley section to external function and call it from `SwitchTrigger`?